From 726344abbbec4beab5525c7890a360f9877d1ffa Mon Sep 17 00:00:00 2001
From: Noel Power <noel.power@novell.com>
Date: Thu, 12 Jan 2012 15:46:52 +0000
Subject: [PATCH 2/8] restore special DATE handling code for SbxValue::Compute
 fdo#44385

restore code that seemed to be erroneously moved as part of the currency enhancement feature
---
 basic/source/sbx/sbxvalue.cxx |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/basic/source/sbx/sbxvalue.cxx b/basic/source/sbx/sbxvalue.cxx
index e21234c..6d9d718 100644
--- a/basic/source/sbx/sbxvalue.cxx
+++ b/basic/source/sbx/sbxvalue.cxx
@@ -1311,9 +1311,6 @@ Lbl_OpIsDouble:
                             else aL.nDouble /= aR.nDouble; break;
                         case SbxPLUS:
                             aL.nDouble += aR.nDouble; break;
-                            // #45465 Date needs with "+" a special handling: forces date type
-                            if( GetType() == SbxDATE || rOp.GetType() == SbxDATE )
-                                aL.eType = SbxDATE;
                         case SbxMINUS:
                             aL.nDouble -= aR.nDouble; break;
                         case SbxNEG:
@@ -1321,6 +1318,9 @@ Lbl_OpIsDouble:
                         default:
                             SetError( SbxERR_NOTIMP );
                     }
+                    // #45465 Date needs with "+" a special handling: forces date type
+                    if( Op == SbxPLUS && (GetType() == SbxDATE || rOp.GetType() == SbxDATE )
+                        aL.eType = SbxDATE;
 
                 }
             }
-- 
1.7.7.6

